alf/config_util.py

```diff
@@ -224,7 +217,26 @@ def config1(config_name, value, mutable=True):
     else:
         config_node = tree
 
-    if config_node.is_used():
+    return config_node
+
+
+def config1(config_name, value, mutable=True, raise_if_used=True):
+    """Set one configurable value.
+
+    Args:
+        config_name (str): name of the config
+        value (any): value of the config
+        mutable (bool): whether the config can be changed later. If the user
+            tries to change an existing immutable config, the change will be
+            ignored and a warning will be generated. You can always change a
+            mutable config. ``ValueError`` will be raised if trying to set a new
+            immutable value to an existing immutable value.
+        raise_if_used (bool): If True, ValueError will be raised if trying to
+            config a value which has already been used.
+    """
+    config_node = _get_config_node(config_name)
+
+    if raise_if_used and config_node.is_used():
```

Comment:
When raise_if_used is False,  maybe we can generate a warning message

Reply:
With the current way using raise_if_used=True inside config_helper, the config is always used, and a warning message will be quite confusing.